[Asset Inventory] Connecting Asset Inventory to the Generic Entity Store

opauloh opened the pull request:
## Summary

It closes #218007

- Updates the asset inventory server to fetch status from the generic entity store. It also creates an asset inventory dataview during onboarding to retrieve entity store data from the current space.

It also ensures that the asset inventory feature is correctly enabled by checking the entity store status and initializing the generic entity engine if needed.

- Updates the asset inventory client to fetch data from the asset inventory dataview. This change removes the direct usage of index patterns and integrates data views for data retrieval and filtering, improving maintainability and flexibility.


- Updates the no data found message to use "asset inventory" rather than "we discover assets".

- Update CODEOWNERS to have the universal flyout owned by cloud security posture team

## Screenshots

### DataView per space

<img width="2057" alt="image" src="https://github.com/user-attachments/assets/e173e535-4819-4093-891b-d7e4f7a1aebd" />


### Generic Flyout

<img width="2053" alt="Screenshot 2025-04-28 at 4 08 13 PM" src="https://github.com/user-attachments/assets/879fb431-6d04-446b-ad98-e3ad47035219" />



### Host Flyout

<img width="2056" alt="Screenshot 2025-04-28 at 4 03 10 PM" src="https://github.com/user-attachments/assets/9b7958ae-d8bf-4815-b217-3c5261c8d781" />



### User Flyout

<img width="2042" alt="image" src="https://github.com/user-attachments/assets/fb51195a-d206-4088-b9d8-9f1c79ac0ae2" />

opauloh committed:
updating status API to fetch data from generic entity store

opauloh committed:
make asset criticality optional

opauloh committed:
Refactors Asset Inventory to use data views

Uses data views for Asset Inventory, which allows for
better index pattern management and filter application.

This change removes the direct usage of index patterns and
integrates data views for data retrieval and filtering,
improving maintainability and flexibility.

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

opauloh committed:
Merge branch 'main' into asset-inventory/connect-generic-entity-store

opauloh committed:
Ensures asset inventory is properly enabled

Ensures that the asset inventory feature is correctly enabled by checking the entity store status and initializing the generic entity engine if needed.

Updates the no data found message to use "asset inventory" rather than "we discover assets".

opauloh committed:
Add universal flyout CODEOWNERS

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/296588)
* Commit: 0f59d2697e5f531c1a321611a3a1b884e9655361

### Failed CI Steps
* [Jest Tests #8](https://buildkite.com/elastic/kibana-pull-request/builds/296588#01967b85-14fd-4177-a5bb-ed1259f24a8d)
* [Jest Tests #8](https://buildkite.com/elastic/kibana-pull-request/builds/296588#01967ba8-de12-4d8f-9589-8c31214c228a)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/296588#01967b85-14fd-4177-a5bb-ed1259f24a8d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/296588/jobs/01967b85-14fd-4177-a5bb-ed1259f24a8d/artifacts/01967ba8-9082-49eb-90c4-8717e2f03894) Jest Tests #<span></span>8 / AssetInventoryDataClient status function returns DISABLED when entity store is not installed
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/296588#01967ba8-de12-4d8f-9589-8c31214c228a) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/296588/jobs/01967ba8-de12-4d8f-9589-8c31214c228a/artifacts/01967bb8-d479-4ad9-a788-26a0f243341d) Jest Tests #<span></span>8 / AssetInventoryDataClient status function returns DISABLED when entity store is not installed

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/296588","number":296588,"commit":"0f59d2697e5f531c1a321611a3a1b884e9655361"}],"number":296588}
buildkite-pr-comment-->

opauloh committed:
reverting disabled status

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/296822)
* Commit: 5c76546ebf29246ce9e5cde2fc26baf718af4292

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/b46e65154ddc7516328cd87b58523809d8e7bfeb) | [after](https://github.com/elastic/kibana/commit/5c76546ebf29246ce9e5cde2fc26baf718af4292) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 9.1MB | 9.1MB | -1.9KB |

<details><summary>Unknown metric groups</summary>

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/b46e65154ddc7516328cd87b58523809d8e7bfeb) | [after](https://github.com/elastic/kibana/commit/5c76546ebf29246ce9e5cde2fc26baf718af4292) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 587 | 586 | -1 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/b46e65154ddc7516328cd87b58523809d8e7bfeb) | [after](https://github.com/elastic/kibana/commit/5c76546ebf29246ce9e5cde2fc26baf718af4292) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 673 | 672 | -1 |


</details>


### History
* :broken_heart: [Build #296588](https://buildkite.com/elastic/kibana-pull-request/builds/296588) failed 0f59d2697e5f531c1a321611a3a1b884e9655361

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/296822","number":296822,"commit":"5c76546ebf29246ce9e5cde2fc26baf718af4292"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/296588","number":296588,"commit":"0f59d2697e5f531c1a321611a3a1b884e9655361"}],"number":296822}
buildkite-pr-comment-->

opauloh committed:
Updates entity type retrieval

Updates entity type retrieval to use `EngineMetadata.Type` for dynamic flyout.

Updates index pattern to support multiple entity types.

Updates entity type definition to include EngineMetadata.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/296955)
* Commit: 6ba38fff61a09ef72f397cd0b5e3d3abf4d200d2

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/b46e65154ddc7516328cd87b58523809d8e7bfeb) | [after](https://github.com/elastic/kibana/commit/6ba38fff61a09ef72f397cd0b5e3d3abf4d200d2) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 9.1MB | 9.1MB | -1.8KB |

<details><summary>Unknown metric groups</summary>

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/b46e65154ddc7516328cd87b58523809d8e7bfeb) | [after](https://github.com/elastic/kibana/commit/6ba38fff61a09ef72f397cd0b5e3d3abf4d200d2) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 587 | 586 | -1 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/b46e65154ddc7516328cd87b58523809d8e7bfeb) | [after](https://github.com/elastic/kibana/commit/6ba38fff61a09ef72f397cd0b5e3d3abf4d200d2) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 673 | 672 | -1 |


</details>


### History
* :green_heart: [Build #296822](https://buildkite.com/elastic/kibana-pull-request/builds/296822) succeeded 5c76546ebf29246ce9e5cde2fc26baf718af4292
* :broken_heart: [Build #296588](https://buildkite.com/elastic/kibana-pull-request/builds/296588) failed 0f59d2697e5f531c1a321611a3a1b884e9655361

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/296955","number":296955,"commit":"6ba38fff61a09ef72f397cd0b5e3d3abf4d200d2"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/296822","number":296822,"commit":"5c76546ebf29246ce9e5cde2fc26baf718af4292"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/296588","number":296588,"commit":"0f59d2697e5f531c1a321611a3a1b884e9655361"}],"number":296955}
buildkite-pr-comment-->

tiansivive reviewed (approved):
LGTM from EA

JordanSh reviewed (approved):
👑
